tests/python/test_identify_tophit_and_contaminants2.py

```diff
@@ -0,0 +1,184 @@
+    with pytest.raises(SystemExit) as pytest_wrapped_e:
+        identify_tophit_and_contaminants2.process_requirements(args)
+    assert pytest_wrapped_e.type == SystemExit
+    assert str(pytest_wrapped_e.value) == 'ERROR: %s is empty' %args[7]
+
+def test_supposed_species():
+    args = [0, os.path.join(data_dir_name, "test_mykrobe_report.json"), os.path.join(data_dir_name, "test_kraken_report.json"), os.path.join(data_dir_name, "assembly_summary_refseq.txt"), 'Randoms', 'no', resources_dir, 'null']
+    with pytest.raises(SystemExit) as pytest_wrapped_e:
+        identify_tophit_and_contaminants2.process_requirements(args)
+    assert pytest_wrapped_e.type == SystemExit
+    assert str(pytest_wrapped_e.value) == 'ERROR: if you provide a species ID, it must be one of either: abscessus|africanum|avium|bovis|chelonae|chimaera|fortuitum|intracellulare|kansasii|tuberculosis'
+
+def test_unmix_myco():
+    args = [0, os.path.join(data_dir_name, "test_mykrobe_report.json"), os.path.join(data_dir_name, "test_kraken_report.json"), os.path.join(data_dir_name, "assembly_summary_refseq.txt"), 'null', 'test', resources_dir, 'null']
+    with pytest.raises(SystemExit) as pytest_wrapped_e:
+        identify_tophit_and_contaminants2.process_requirements(args)
+    assert pytest_wrapped_e.type == SystemExit
+    assert str(pytest_wrapped_e.value) == 'ERROR: \'unmix myco\' should be either \'yes\' or \'no\''
+
+def test_unmatched_ids():
+    # mismatched mykrobe and kraken
+    args = [0, os.path.join(data_dir_name, "mismatched_mykrobe_report.json"), os.path.join(data_dir_name, "test_kraken_report.json"), os.path.join(data_dir_name, "assembly_summary_refseq.txt"), 'null', 'no', resources_dir, 'null']
+    with pytest.raises(SystemExit) as pytest_wrapped_e:
+        identify_tophit_and_contaminants2.process_requirements(args)
+    assert pytest_wrapped_e.type == SystemExit
+    assert str(pytest_wrapped_e.value) == "ERROR: the sample IDs of %s and %s are mismatched" %(os.path.join(data_dir_name, "mismatched_mykrobe_report.json"), os.path.join(data_dir_name, "test_kraken_report.json"))
+
+    # test mismatched previous species id
+    args = [0, os.path.join(data_dir_name, "test_mykrobe_report.json"), os.path.join(data_dir_name, "test_kraken_report.json"), os.path.join(data_dir_name, "assembly_summary_refseq.txt"), 'null', 'no', resources_dir, os.path.join(data_dir_name, "mismatched_species_in_sample.json")]
+    with pytest.raises(SystemExit) as pytest_wrapped_e:
+        identify_tophit_and_contaminants2.process_requirements(args)
+
+    assert pytest_wrapped_e.type == SystemExit
+    assert str(pytest_wrapped_e.value) == "ERROR: sample ID of the previous species JSON (%s) does not match the sample ID we have from the Kraken and Mykrobe reports (%s)" %(args[7], os.path.join(data_dir_name, 'test'))
```

Comment:
This line results in an AssertionError when running with pytest. Looks like ```os.path.join(data_dir_name, 'test')``` needs to be replaced with ```test```

Reply:
Hello Anna,

Thanks for the code review and suggestion. I have fixed the issue now following your suggestion and pushed the fix into the pull request here with an extra commit.

Seems like it worked fine on my local paths but was not working correctly with different routes.

Also, I have found out that it was the same issue in the original perl scripts, which resulted in finding incorrect report IDs. I have now adjusted regex for getting report IDs in order to provide a more general fix. 

Thanks,
Maxim